### PR TITLE
buffer: api change

### PR DESCRIPTION
**Pull request by kaspersky**
Fix issue #1660

### CODE REVIEW

**Comment by cabelitos** [comment]
+1

File: src/lib/datatypes/include/sol-buffer.h [hunk 1]
```diff
@@ -894,7 +894,7 @@ int sol_buffer_ensure_nul_byte(struct sol_buffer *buf);
  * @note the buffer keeps its capacity after this function, it means,
  * the data is not released. If that is wanted, one should call @ref sol_buffer_trim
  */
-int sol_buffer_remove_data(struct sol_buffer *buf, size_t size, size_t offset);
+int sol_buffer_remove_data(struct sol_buffer *buf, size_t offset, size_t size);
```
Comment (otaviobp) [inline src/lib/datatypes/include/sol-buffer.h]: Please, change the order of size and offset in comments too. Doxygen works fine, but it is easier to read in .h files if the order of parameters is correct.

File: src/test/test-buffer.c [hunk 2]
```diff
@@ -781,23 +781,23 @@ test_remove_data(void)
     ASSERT_INT_EQ(buf.used, strlen("ABCDEFGHI"));
     ASSERT_STR_EQ(buf.data, "ABCDEFGHI");
 
-    err = sol_buffer_remove_data(&buf, strlen("ABC"), 0);
+    err = sol_buffer_remove_data(&buf, 0, strlen("ABC"));
     ASSERT_INT_EQ(err, 0);
     ASSERT_INT_EQ(buf.used, strlen("DEFGHI"));
 
-    err = sol_buffer_remove_data(&buf, strlen("GHI"), 3);
+    err = sol_buffer_remove_data(&buf, 0, strlen("GHI"));
```
Comment (otaviobp) [inline src/test/test-buffer.c]: why are you using 0 instead of 3?

File: src/lib/datatypes/include/sol-buffer.h (hunk 1 above)
Comment (kaspersky) [inline src/lib/datatypes/include/sol-buffer.h]: Will do.

File: src/test/test-buffer.c (hunk 2 above)
Comment (kaspersky) [inline src/test/test-buffer.c]: Mistake, I will fix it.

**Pull request closed without merging**